File: 02.rb
def fib(a,b,limit,&block)
  yield a
  fib(b, (a+b), limit, &block) unless (b) > limit
end

total = 0
fib(1,2,4000000) {|x| total += x if (x%2 == 0)}
puts total

File: 02.py
def fib(limit, nums=[1,2]):
	next = nums[-1]+nums[-2]
	if next > limit:
		return nums
	else:
		nums.append(next)
		return fib(limit, nums)
		
print sum([n for n in fib(4000000) if n % 2 == 0])
